tests/mocha/widget_div_test.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+suite('WidgetDiv', function() {
+
+  suite('positionWithAnchor', function() {
+    function makeBBox(left, top, width, height) {
+      return {
+        left: left,
+        right: left + width,
+        top: top,
+        bottom: top + height,
+        width: width,
+        height: height
+      };
+    }
+
+    setup(function() {
+      Blockly.WidgetDiv.createDom();
+      this.viewportBBox = makeBBox(0, 0, 1000, 1003);
+      this.widgetSize = {
+        width: 100,
+        height: 102
+      };
+      this.anchorSize = {
+        width: 90,
+        height: 91
+      };
+
+      this.testWidgetPosition = function(
```

Comment:
Why not define the function similar to how makeBBox?

Reply:
I wanted to be able to access `this` for `viewportBBox` and `widgetSize` and I needed to define it differently for that.